electronics_lib/Passives.py

```diff
@@ -314,6 +297,195 @@ def derated_capacitance(row: Dict[str, Any]) -> Tuple[float, float]:
       raise ValueError(f"no single capacitors in ({capacitance}) F")
 
 
+class SmtCeramicCapacitorGeneric(Capacitor, FootprintBlock, GeneratorBlock):
+
+  SINGLE_CAP_MAX = 22e-6
+
+  @init_in_parent
+  def __init__(self, *args, **kwargs):
+    super().__init__(*args, **kwargs)
+
+    self.footprint_spec = self.Parameter(StringExpr(""))
+    self.derating_coeff = self.Parameter(FloatExpr(1))
+
+    # Default to be overridden on a per-device basis
+    self.single_nominal_capacitance = self.Parameter(RangeExpr((0, self.SINGLE_CAP_MAX * 1.25)))  # maximum capacitance in a single part
+
+    self.generator(self.select_capacitor_no_prod_table, self.capacitance, self.voltage, self.single_nominal_capacitance,
+                   self.footprint_spec, self.derating_coeff)
+
+    # Output values
+    self.selected_capacitance = self.Parameter(RangeExpr())
+    self.selected_derated_capacitance = self.Parameter(RangeExpr())
```

Comment:
Might be clearer as selected_nominal_capacitance and selected_effective_capacitance?
Derating I think is more often used in talking about parts having a lower than labeled capacitance under load, whereas we're kind of applying the inverse here.

I'm also not sure the second is useful (pre-derating effective capacitance), because you'd need to then apply derating from the selected E-series and parallel caps to get the correct value.